src/mysql_encode.erl

```diff
@@ -23,6 +23,12 @@ encode(String) when is_list(String) ->
     encode(unicode:characters_to_binary(String));
 encode(Bitstring) when is_bitstring(Bitstring) ->
     ["b'", [ case B of 0 -> $0; 1 -> $1 end || <<B:1>> <= Bitstring ], $'];
+encode({decimal, Num}) when is_float(Num); is_integer(Num) ->
+    encode(Num);
+encode({decimal, Str}) when is_binary(Str); is_list(Str) ->
+    %% Simple injection block
+    nomatch = re:run(Str, <<"[^0-9\\.eE+-]">>),
```

Comment:
The `.` loses its special meaning when used in a character class, no need to escape it.

Also, while it is unlikely that this regexp ever changes, it might be good to add a comment to remember that the `-` must be the last character in a character class, otherwise it becomes a range operator. And/or escape the `-` just in case.

Reply:
Good ideas! I will change.